File: leetcode/solutions/0053.MaximumSubarray/maxSubArray_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import maxSubArray from "./maxSubArray.ts";

Deno.test("0053. Maximum Subarray", () => {
  type test = [number[], number];
  const tests: test[] = [
    [[-1], -1],
    [[-3, -2, -3, -4], -2],
    [[0, 0, 0, 0, 0], 0],
    [[1, 1, 1, 1, 1], 5],
    [[1, -1, 1, -1, 1], 1],
    [[1, -1, 2, -1, 2], 3],
    [[-2, 1, -3, 4, -1, 2, 1, -5, 4], 6],
  ];

  for (const [arg1, want] of tests) {
    assertEquals(maxSubArray(arg1), want);
  }
});

File: leetcode/solutions/0053.MaximumSubarray/maxSubArray_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_0053_MaximumSubarray(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  []int
		out int
	}{
		{[]int{-1}, -1},
		{[]int{1}, 1},
		{[]int{-3, -2, -3, -4}, -2},
		{[]int{0, 0, 0, 0, 0}, 0},
		{[]int{1, 1, 1, 1, 1}, 5},
		{[]int{1, -1, 1, -1, 1}, 1},
		{[]int{1, -1, 2, -1, 2}, 3},
		{[]int{-2, 1, -3, 4, -1, 2, 1, -5, 4}, 6},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := maxSubArray(tt.in)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}
